scripts/west_commands/runners/intel_adsp.py

```diff
@@ -124,6 +134,15 @@ def flash(self, **kwargs):
         self.run_cmd = ([f'{self.cavstool}','-s', f'{self.remote_host}', f'{send_bin_fw}'])
         self.log_cmd = ([f'{self.cavstool}','-s', f'{self.remote_host}', '-l'])
 
+        # Specify the ports of the req and log service if needed.
+        if self.port_req:
+            self.run_cmd.append('--port-req')
+            self.run_cmd.append(f'{self.port_req}')
+
+        if self.port_log:
+            self.log_cmd.append('--port-log')
+            self.log_cmd.append(f'{self.port_log}')
+
```

Comment:
> Adding one leads to a never-ending treadmill of adding two, three, four, [...] .

Agree 100%

> like the way `--tool-opt`

I haven't look at this one in particular but the problem with these is always the same: whitespace and quoting. Example:

```
# is --opt2 for west or for subtool?
west  --subtool-opt --opt1 --opt2

# OK now it is for west. But then...
west  --subtool-opt '--opt1 --opt2'

# Now let's try this
west --subtool-opt '--opt1="-DFU1=bar1 -DFU2=bar2" --opt2'

subtool ERROR: FU1 cannot be 'bar1 -DFU2=bar2'
```
etc.

The only reliable solution on the CLI is `--` but it allows only one subtool.

There is only so much a CLI can "scale". Past that point configuration files are required.

Reply:
> I haven't look at this one in particular but the problem with these is always the same: whitespace and quoting.

I think you will find that the implelmentations I pointed to solve both of these problems:

```py
parser.add_argument('--tool-opt', default=[], action='append', ...)
```

You can basically give as many of these as you want, and the result will be a python list which can contain arbitrary characters, including whitespace characters.

Reply:
Thanks @mbolivar-nordic ! So just to make absolutely sure there is zero misunderstanding:
```
west ... --tool-opt '-DKEYWORD1=value with some white space' --tool-opt --opt2
```
Correct? A thumbs up should be enough :-)

Reply:
> west ... --tool-opt '-DKEYWORD1=value with some white space' --tool-opt --opt2

Of course this does not work because... "thank you `argparse`"? Example:

```
west update --fetch-opt --depth=5
west update: error: argument -o/--fetch-opt: expected one argument
```
Workaround:
```
west update --fetch-opt=--depth=5
```

Infamous https://bugs.python.org/issue9334 "argparse does not accept options taking arguments beginning with dash (regression from optparse)"

Reply:
Thank very much for suggestions from @marc-hb @mbolivar-nordic , got learned again. I updated it according your recommendation.